modules/Movecraft/src/main/java/net/countercraft/movecraft/mapUpdater/update/EntityUpdateCommand.java

```diff
@@ -52,12 +82,16 @@ public Entity getEntity() {
     @Override
     public void doUpdate() {
         Location playerLoc = entity.getLocation();
+        Location location = new Location(world, playerLoc.getX() + x, playerLoc.getY() + y, playerLoc.getZ() + z, playerLoc.getYaw() + yaw, playerLoc.getPitch() + pitch);
         if (!(entity instanceof Player) || yaw > .01 || pitch > .01 || yaw < -.01 || pitch < -.01) {
-            entity.teleport(new Location(entity.getWorld(), x + playerLoc.getX(),y + playerLoc.getY(),z + playerLoc.getZ(),yaw + playerLoc.getYaw(),pitch + playerLoc.getPitch()));
+            entity.teleport(location);
             return;
         }
         //Movecraft.getInstance().getWorldHandler().addPlayerLocation((Player) entity,x,y,z,yaw,pitch);
-        TeleportUtils.teleport((Player) entity, new Location(entity.getWorld(), playerLoc.getX() + x, playerLoc.getY() + y, playerLoc.getZ() + z));
+        TeleportUtils.teleport((Player) entity, location);
+        if (sound != null) {
```

Comment:
I don't think sound playing belongs in this, it should only be teleporting players.

Reply:
I disagree, the class is called EntityUpdateCommand not EntityTeleportCommand, plus, the sound should play right after the player is teleported so this would be the most reasonable place to put it.